aspnetcore/blazor/components/prerender.md

```diff
@@ -17,358 +17,134 @@ uid: blazor/components/prerender
+            environmentName = env.Environment;
         }
-
-        // Call at the end to avoid a potential race condition at app shutdown
-        persistingSubscription = ApplicationState.RegisterOnPersisting(PersistCount);
     }
-
-    private Task PersistCount()
-    {
-        ApplicationState.PersistAsJson(nameof(currentCount), currentCount);
-
-        return Task.CompletedTask;
-    }
-
-    private void IncrementCount() => currentCount++;
-
-    void IDisposable.Dispose() => persistingSubscription.Dispose();
 }
 ```
 
-When the component executes, `currentCount` is only set once during prerendering. The value is restored when the component is rerendered. The following is example output.
+However, the preceding approach isn't useful if your logic requires a value during prerendering.
 
-> [!NOTE]
-> If the app adopts [interactive routing](xref:blazor/fundamentals/routing#static-versus-interactive-routing) and the page is reached via an internal [enhanced navigation](xref:blazor/fundamentals/routing#enhanced-navigation-and-form-handling), prerendering doesn't occur. Therefore, you must perform a full page reload for the component to see the following output. For more information, see the [Interactive routing and prerendering](#interactive-routing-and-prerendering) section.
-
-> :::no-loc text="info: BlazorSample.Components.Pages.PrerenderedCounter3[0]":::  
-> :::no-loc text="      currentCount set to 96":::  
-> :::no-loc text="info: BlazorSample.Components.Pages.PrerenderedCounter3[0]":::  
-> :::no-loc text="      currentCount restored to 96":::
-
-:::moniker-end
+You can also avoid the problem if you [disable prerendering](#disable-prerendering) for the component, but that's an extreme measure to take in many cases that may not meet your component's specifications.
 
-:::moniker range="< aspnetcore-10.0"
+There are a three approaches that you can take to address this scenario. The following are listed from most recommended to least recommended:
 
-To preserve prerendered state, decide what state to persist using the <xref:Microsoft.AspNetCore.Components.PersistentComponentState> service. <xref:Microsoft.AspNetCore.Components.PersistentComponentState.RegisterOnPersisting%2A?displayProperty=nameWithType> registers a callback to persist the component state during prerendering. The state is retrieved when the component renders interactively. Make the call at the end of initialization code in order to avoid a potential race condition during app shutdown.
+* *Recommended* for shared framework services: For shared framework services that merely aren't registered server-side in the main project, register the services in the main project, which makes them available during prerendering. For an example of this scenario, see the guidance for <xref:System.Net.Http.HttpClient> services in <xref:blazor/call-web-api?pivots=webassembly#client-side-services-for-httpclient-fail-during-prerendering>.
```

Comment:
The anchor linked to in the call a web API doc doesn't seem to exist. Were you trying to link to xref:blazor/call-web-api#service-abstractions-for-web-api-calls instead? That section recommends using a service abstraction, which is good. But that doesn't seem related to shared framework services like IWebAssemblyHostEnvironment unless you want to create your own interface abstraction for getting the environment.